Tests/Library/Client.tests.ts

```diff
@@ -325,12 +372,133 @@ describe("Library/Client", () => {
                 assert.deepEqual(args[0][0].baseData.properties, properties);
             });
         });
+
+        describe("#trackDependencyRequest()", () => {
+            var clock: SinonFakeTimers;
+
+            before(() => {
+                clock = sinon.useFakeTimers();
+            });
+
+            after(() => {
+                clock.restore();
+            });
+
+            it("should not crash with invalid input", () => {
+                invalidInputHelper("trackDependencyRequest");
+            });
+
+            it('should track request with correct data from request options', () => {
+                trackStub.reset();
+                clock.reset();
+                client.trackDependencyRequest({
+                        host: 'bing.com',
+                        path: '/search?q=test'
+                    },
+                    <any>request, properties);
+
+                // response event was not emitted yet
+                assert.ok(trackStub.notCalled);
+
+                // emit response event
+                clock.tick(10);
+                request.emitResponse();
+                assert.ok(trackStub.calledOnce);
+                var args = trackStub.args;
+                assert.equal(args[0][0].baseType, "Microsoft.ApplicationInsights.RemoteDependencyData");
```

Comment:
It would be more readable and easier to edit in future if args[0][0] was cached into a var instead of repeating the dereference each time.